x/twap/README.md

```diff
@@ -12,6 +12,28 @@ Notice that the latest price `p_n` isn't used, as it has lasted for a time inter
 To illustrate with an example, given the sequence: `(0s, $1), (4s, $6), (5s, $1)`, the arithmetic mean TWAP is: 
 $$\frac{\$1 * (4s - 0s) + \$6 * (5s - 4s)}{5s - 0s} = \frac{\$10}{5} = \$2$$
 
+## Geometric mean TWAP
+
+While the arithmetic mean TWAPs are much more widely used, they should theoretically be less accurate in measuring a geometric Brownian motion process (which is how price movements are usually modeled)
+
+Arithmetic TWAP tends to overweight higher prices relative to lower ones.
+
+Therefore, we also support a geometric mean TWAP.
+
+The core functionality stays similar to the arithmetic mean TWAP. However, instead of computing the geometric mean TWAP naively, we use the following property:
```

Comment:
First the naive should be defined

Then how we compute it =p

Reply:
[36841f5](https://github.com/osmosis-labs/osmosis/pull/3420/commits/36841f5ee92e1370aeb98ced873b9faa7b3d8ceb)